linkedlist/LinkedList.js

```diff
@@ -30,39 +28,69 @@ export class LinkedList {
      * @returns {any | null}
      */
     get head() {
-        // TODO: implement
+        return this.#headNode?.value ?? null;
     }
 
     /**
      * Returns the last value in the list.
      * @returns {any | null}
      */
     get tail() {
-        // TODO: implement
+        if (this.#headNode === null) {
+            return null;
+        }
+
+        let temp = this.#headNode;
+        while (temp.nextNode !== null) {
+            temp = temp.nextNode;
+        }
+        return temp.value;
     }
 
     /**
      * Returns the number of values in the list.
      * @returns {number}
      */
     get size() {
-        // TODO: implement
+        if (this.#headNode === null) {
+            return 0;
+        }
+
+        let length = 1;
+        let temp = this.#headNode;
+        while (temp.nextNode !== null) {
+            length++;
+            temp = temp.nextNode;
+        }
+        return length;
     }
 
     /**
      * Adds the given value to the end of the list.
      * @param {any} value 
      */
     append(value) {
-        // TODO: implement
+        if (this.#headNode === null) {
+            this.#headNode = new Node(value);
+        } else {
+            let temp = this.#headNode;
+            while (temp.nextNode !== null) {
+                temp = temp.nextNode;
+            }
+            temp.nextNode = new Node(value);
+        }
     }
 
     /**
      * Adds the given value to the start of the list.
      * @param {any} value 
      */
     prepend(value) {
-        // TODO: implement
+        if (this.#headNode === null) {
```

Comment:
Can you do this without the `if`/`else`?

Reply:
Yes.